### PR TITLE
Added release workflow

### PR DESCRIPTION
**Pull request by lupengamzn**
*Description of changes:*

Added workflow to release assets and generate tag on this repo

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: 'v${{ github.event.inputs.version }}'
+          release_name: 'Release ${{ github.event.inputs.version }}'
+          body: 'Please refer [change-log](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/CHANGELOG.md) for more details'
+          draft: false
```
Comment (bhautikpip) [inline .github/workflows/release.yml:26]: we should set this flag as `true`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Checkout master branch
+        uses: actions/checkout@v2
+
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: 'v${{ github.event.inputs.version }}'
```
Comment (bhautikpip) [inline .github/workflows/release.yml:23]: We should add capital `V` here? looks like all the .NET tag follows V2.9.0, V2.8.0 and so on from [here](https://github.com/aws/aws-xray-sdk-dotnet/tags)

File: .github/workflows/release.yml (hunk 1 above)
Comment (lupengamzn) [reply .github/workflows/release.yml]: Updated

**Review by bhautikpip — APPROVED** [review]
LGTM

**Pull request merged**